src/main/java/BattleshipModel.java

```diff
@@ -39,9 +40,9 @@ public void aiFire(Board target, Coordinate shot){
 
     }
```

Comment:
Seems slightly unnecessary to split this up into two function when you can use the same function for both player and ai by passing which board you're firing at, especially since the coordinate to fire at is calculated elsewhere since it's passed in as a parameter.